### DISCUSSION TITLE
Have two agents on two machines connected to my gateway; clients on either agent machine cannot see the other agent

### DISCUSSION THREAD

**Discussion by ejheil** [narrowlink/narrowlink#94] [Q&A] [answered]
As mentioned in the title -- my gateway now has two machines with two agents successfully connected to it; however when I run `narrowlink list` on either machine, it can only see its own agent.

I'm assuming that an agent on one machine needs to see the other agent on another machine in order (for example) to make an ssh connection to it.

**Comment by ejheil** [comment]
Having read this discussion https://github.com/narrowlink/narrowlink/discussions/11

I should point out that machine A's client and agent have the same name and ID, and machine B's client and agent have a separate name and ID.

I think that's as it should be?  A unique name & id for each agent/client pair?

**Reply by ejheil** [reply]
Aha, I figured out my problem.  Different agents need different agent names, but to see each other they have to belong to the same user (have the same uid).

Once I made all the agents' uids the same they could see each other.

**Reply by SajjadPourali** [reply, marked as answer]
Hello,

Exactly, UID is considered a user space (or zone), mainly designed to support multiple users. Only clients can reach the agents if they are in the same user space (zone). On the other hand, names are just identifiers to separate two entities from each other. You don't need to use the same name for each machine, and agents and clients must not use the same name; otherwise, the previously connected one will be kicked.
As a result, to see your agents, we have to have the same UID for all but a unique name for each.

I am glad that you solved the issue. Please don't hesitate to tell me if any further elaboration is required.

**Reply by ejheil** [reply]
Thank you!  I hadn't really understood well what was meant by "for each user" when it appeared in the config file documentation.

Maybe because I started out by doing a web server publish, where "users" weren't as salient a concept, and only later came to try and use the narrow link client to talk to other agents.
